File: app/src/main/java/com/dboy/dialog/java/viewBinding/BaseDialogFragment.java
package com.dboy.dialog.java.viewBinding;

import android.app.Dialog;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.util.DisplayMetrics;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;
import android.view.WindowManager;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.DialogFragment;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.fragment.app.FragmentManager;
import androidx.viewbinding.ViewBinding;

import com.dboy.dialog.MyDialog;

/**
 * @author  Dboy
 *  viewBinding封装的baseDialog
 */
public abstract class BaseDialogFragment<T extends ViewBinding> extends DialogFragment implements IDialog<T> {

    protected T mRootView;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mRootView = getRootViewBinding(inflater, container);
        return mRootView.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        setDialogWidthSize(getWidthPercentSize(), getHeightPercentSize());
        initViewAndData(savedInstanceState);
    }

    private void setDialogWidthSize(float widthPercent, float heightPercent) {
        DisplayMetrics dm = new DisplayMetrics();
        FragmentActivity activity = getActivity();
        Dialog dialog = getDialog();
        if (activity != null && dialog != null) {
            activity.getWindowManager().getDefaultDisplay().getMetrics(dm);
            Window window = dialog.getWindow();
            if (window != null) {
                window.getDecorView().setPadding(0, 0, 0, 0);
                WindowManager.LayoutParams lp = window.getAttributes();

                lp.width = widthPercent == 0 ? WindowManager.LayoutParams.WRAP_CONTENT : widthPercent == 1 ? WindowManager.LayoutParams.MATCH_PARENT : (int) (dm.widthPixels * widthPercent);
                lp.height = heightPercent == 0 ? WindowManager.LayoutParams.WRAP_CONTENT : heightPercent == 1 ? WindowManager.LayoutParams.MATCH_PARENT : (int) (dm.heightPixels * heightPercent);

                window.setAttributes(lp);
                window.setBackgroundDrawable(new ColorDrawable());
            }
        }
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        return new MyDialog(requireContext(), getTheme());
    }

    public void show(FragmentActivity fragmentActivity) {
        show(fragmentActivity.getSupportFragmentManager());
    }

    public void show(Fragment fragment) {
        show(fragment.getChildFragmentManager());
    }
    @Override
    public void show(@NonNull FragmentManager manager) {
        try {
            manager.beginTransaction().remove(this).commitAllowingStateLoss();
            super.show(manager, getClass().getSimpleName());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/dboy/dialog/kotlin/viewBinding/BaseDialogFragment.kt
package com.dboy.dialog.kotlin.viewBinding

import android.app.Dialog
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.util.DisplayMetrics
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.WindowManager
import androidx.annotation.FloatRange
import androidx.fragment.app.DialogFragment
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentManager
import androidx.viewbinding.ViewBinding
import com.dboy.dialog.MyDialog
/**
 * @author  Dboy
 *  viewBinding封装的baseDialog
 */
open abstract class BaseDialogFragment<T : ViewBinding> : DialogFragment(), IDialog<T> {

    lateinit var mRootView: T

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        mRootView = getRootViewBinding(inflater, container)
        return mRootView.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setDialogWidthSize(getWidthPercentSize(), getHeightPercentSize())
        initViewAndData(savedInstanceState)
    }

    open fun show(activity: FragmentActivity) {
        show(activity.supportFragmentManager)
    }

    open fun show(fragment: Fragment) {
        show(fragment.childFragmentManager)
    }
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        return MyDialog(requireContext(),theme)
    }
    override fun show(manager: FragmentManager) {
        try {
            manager.beginTransaction().remove(this).commitAllowingStateLoss()
            super.show(manager, javaClass::getSimpleName.toString())
        } catch (e: Exception) {
        }
    }

    protected fun setDialogWidthSize(widthPercent: Float, heightPercent: Float) {
        val dm = DisplayMetrics()
        if (activity != null && dialog != null) {
            activity?.windowManager?.defaultDisplay?.getMetrics(dm)
            val window = dialog?.window
            if (window != null) {
                window.decorView.setPadding(0, 0, 0, 0)
                val lp = window.attributes
                lp.width =
                    if (widthPercent == 0f) WindowManager.LayoutParams.WRAP_CONTENT else if (widthPercent == 1f) WindowManager.LayoutParams.MATCH_PARENT else (dm.widthPixels * widthPercent).toInt()
                lp.height =
                    if (heightPercent == 0f) WindowManager.LayoutParams.WRAP_CONTENT else if (heightPercent == 1f) WindowManager.LayoutParams.MATCH_PARENT else (dm.heightPixels * heightPercent).toInt()
                window.attributes = lp
                window.setBackgroundDrawable(ColorDrawable())
            }
        }
    }

}

interface IDialog<T : ViewBinding> {

    /**
     * 获取binding
     */
    fun getRootViewBinding(inflater: LayoutInflater, container: ViewGroup?): T

    /**
     * 初始化视图和数据
     */
    fun initViewAndData(savedInstanceState: Bundle?)

    /**
     * @return dialog的宽度比例 0=内容包裹，1=匹配父布局， 0.0~1.0 按屏幕比例计算
     */
    @FloatRange(from = 0.0, to = 1.0)
    fun getWidthPercentSize(): Float

    /**
     * @return dialog的高度比例 0=内容包裹， 1=匹配父布局， 0.0~1.0 按屏幕比例计算
     */
    @FloatRange(from = 0.0, to = 1.0)
    fun getHeightPercentSize(): Float

    /**
     * 简化展示
     */
    fun show(manager: FragmentManager)
}